json_spec/README.md

```diff
@@ -1,16 +1,2589 @@
+</table>
+
+##### Example
+
+```json
+{
+  "testStepArtifact": {
+    "testStepEnd": {
+      "status": "COMPLETE"
+    },
+    "testStepId": "22"
+  },
+  "sequenceNumber": 200,
+  "timestamp": "2022-07-25T04:54:41.292679Z"
+}
+```
+
+
+#### TestStepStart
+
+##### Description
+
+The [_TestStepStart_](#teststepstart) [**artifact**](#def-artifact) contains all the information relating to the beginning of a new phase, task, step, or area of focus for a [**diagnostic**](#def-diagnostic).
+
+##### Attributes
+
+<table>
+    <tr>
+        <td><strong>Attribute</strong></td>
+        <td><strong>Type</strong></td>
+        <td><strong>Required</strong></td>
+        <td><strong>Description</strong></td>
+    </tr>
+    <tr>
+        <td><em>name</em></td>
+        <td>string</td>
+        <td><strong>Yes</strong></td>
+        <td>Name of the <a href="#def-teststep"><strong>test step</strong></a></td>
+    </tr>
+</table>
+
+##### Example
+
+```json
+{
+  "testStepArtifact": {
+    "testStepStart": {
+      "name": "intranode-bandwidth-check"
+    },
+    "testStepId": "22"
+  },
+  "sequenceNumber": 200,
+  "timestamp": "2022-07-25T04:56:42.249367Z"
+}
+```
+
+
+#### Validator
+
+##### Description
+
+A [_validator_](#validator) is the description of a comparison or evaluation of a [_measurement_](#measurement) or [_measurement series element_](#measurementserieselement) against a criteria to determine if the [_measurement_](#measurement) is [**conformant**](#def-nonconformance) or within expected tolerances. [_Validators_](#validator) are polymorphic and can support multiple value types, although the data types on both sides of the evaluation shall have the same type, or the behavior is not supported.
+
+##### Attributes
+
+<table>
+    <tr>
+        <td><strong>Attribute</strong></td>
+        <td><strong>Type</strong></td>
+        <td><strong>Required</strong></td>
+        <td><strong>Description</strong></td>
+    </tr>
+    <tr>
+        <td><em>name</em></td>
+        <td>string</td>
+        <td>No</td>
+        <td>The name of the comparison being performed (i.e. checksum verification)</td>
+    </tr>
+    <tr>
+        <td><em>type</em></td>
+        <td>JSON: string<br/>Message: enum <a href="#validatortype">ValidatorType</a></td>
+        <td><strong>Yes</strong></td>
+        <td>A supported validator/comparison type.</td>
+    </tr>
+    <tr>
+        <td><em>value</em></td>
+        <td>JSON: string, number, boolean</td>
+        <td><strong>Yes</strong></td>
+        <td>The value to use on the right side of the arithmetic comparison.</td>
+    </tr>
+    <tr>
+        <td><em>metadata</em></td>
+        <td>JSON: object<br/>Message: <a href="#metadata"><em>Metadata</em></a></td>
+        <td>No</td>
+        <td>Optional <a href="#metadata"><em>metadata</em></a> about the <a href="#validator"><em>validator</em></a>.</td>
+    </tr>
+</table>
+
+---
+# TODO: duplicated with section on enums (this one is left uncorrected for links)
```

Comment:
What is the proposal here, remove this?  I think I would keep it for readability/parseability.  (Not sure what the 'uncorrected' comment meant as well).

Reply:
well, i can leave it as is for the time being; removing it later on in either of the 2 places will not change the text of the spec. but the point was that it's duplicated in the enums section at beginning of file, and here. at the very least, this might lead to drift because of forgetting to update both places on changes

Reply:
done. please check/resolve